cli/src/cmd/utils.rs

```diff
@@ -151,6 +151,16 @@ impl From<RetryArgs> for Retry {
     }
 }
 
+/// Returns error if constructor has arguments.
+pub fn ensure_clean_constructor(abi: &Abi) -> eyre::Result<()> {
+    if let Some(constructor) = &abi.constructor {
+        if !constructor.inputs.is_empty() {
+            eyre::bail!("Contract constructor should have no arguments. Add those arguments to  `run(...)` instead, and call it with `--sig run(...)`.");
```

Comment:
"Constructor arguments are unsupported. Use `run()` function instead and provide the `--sig`... argument"

Reply:
afaik, people who put arguments on the constructor want to be able to call a script with arguments. This error message doesn't seem to help in that regard. They might think it's unsupported to do `run(uint a)`, which is not.

Reply:
I see, totally agree, sgtm!